plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -150,22 +152,42 @@ private void buildTransportForUrl(String url) throws URISyntaxException {
   }
 
   private Proxy createProxy(String url) throws URISyntaxException {
-    Preconditions.checkNotNull(proxyService, "proxyService is null");
     Preconditions.checkNotNull(googleApiUrls, "googleApiUrl is null");
+    Preconditions.checkArgument(!url.startsWith("http://"), "http is not supported schema");
+
+    if (proxyService == null) {
+      return Proxy.NO_PROXY;
+    }
 
     IProxyData[] proxyData = proxyService.select(new URI(url));
     for (final IProxyData iProxyData : proxyData) {
-      if (IProxyData.HTTPS_PROXY_TYPE.equals(iProxyData.getType())) {
-        return new Proxy(Type.HTTP, new InetSocketAddress(iProxyData.getHost(),
-                                                          iProxyData.getPort()));
+      switch (iProxyData.getType()) {
+        case IProxyData.HTTPS_PROXY_TYPE:
+          return new Proxy(Type.HTTP, new InetSocketAddress(iProxyData.getHost(),
+                                                            iProxyData.getPort()));
+        case IProxyData.SOCKS_PROXY_TYPE:
+          return new Proxy(Type.SOCKS, new InetSocketAddress(iProxyData.getHost(),
+                                                             iProxyData.getPort()));
+        default:
```

Comment:
This code isn't handling a proxy directive, and we should at least log something.